test/common.py

```diff
@@ -2106,24 +2106,27 @@ def reftest(self, expected, manually_trigger=False):
       setupRefTest();
 ''' % (reporting, basename, int(manually_trigger)))
 
-  def compile_btest(self, args, reporting=Reporting.FULL):
+  def compile_btest(self, filename, args, reporting=Reporting.FULL):
     # Inject support code for reporting results. This adds an include a header so testcases can
     # use REPORT_RESULT, and also adds a cpp file to be compiled alongside the testcase, which
     # contains the implementation of REPORT_RESULT (we can't just include that implementation in
     # the header as there may be multiple files being compiled here).
     if reporting != Reporting.NONE:
       # For basic reporting we inject JS helper funtions to report result back to server.
-      args += ['-DEMTEST_PORT_NUMBER=%d' % self.port,
-               '--pre-js', test_file('browser_reporting.js')]
+      args += ['--pre-js', test_file('browser_reporting.js'),
+               '-include', test_file('report_result.h')]
```

Comment:
Why did this change?

Reply:
Because `report_result.c` is now compiled separately and linked in as an object file.    We need to do this because the test itself might be C or C++ but `report_result.c` is C.. we can't use a single compile command to compile both C and C++ without generating the `clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]` warning.

Reply:
Oops, you are right, the `-include` part should not have moved outside the condition.  Reverted that.